infrastructure/tests/test_data_import.py

```diff
@@ -6,14 +6,85 @@
 
 
+def mock_project_row():
+    """This is the project data that mock_existing_project_row will update"""
+    mock_data = { 'Function': 'Administrative and Corporate Support',
+        'Project Description': 'P-CNIN FURN & OFF EQUIP',
+        'Project Number': 'PC002003005_00002',
+        'Type': 'New',
+        'MTSF Service Outcome': 'An efficient, effective and development-oriented public service',
+        'IUDF': 'Growth',
+        'Own Strategic Objectives': 'OWN MUNICIPAL STRATEGIC OBJECTIVE',
+        'Asset Class': 'Furniture and Office Equipment',
+        'Asset Sub-Class': '',
+        'Ward Location': 'Administrative or Head Office',
+        'GPS Longitude': '0', 'GPS Latitude': '0',
+        'Audited Outcome 2017/18': 2000.0,
+        'Full Year Forecast 2018/19': 3000.0,
+        'Budget year 2019/20': 4000.0,
+        'Budget year 2020/21': 5000.0,
+        'Budget year 2021/22': 6000.0
+    }
+    yield mock_data
+
+def mock_data_existing_project_row():
+    """This should update the project from mock_project_row()"""
+    mock_data = { 'Function': 'Administrative and Corporate Support',
+        'Project Description': 'P-CNIN FURN & OFF EQUIP',
+        'Project Number': 'PC002003005_00002',
+        'Type': 'Renewal',
+        'MTSF Service Outcome': 'A project update',
+        'IUDF': 'Governance',
+        'Own Strategic Objectives': 'TO BE CORRECTED',
+        'Asset Class': 'Furniture and Office',
+        'Asset Sub-Class': 'Equipment',
+        'Ward Location': 'Head Office',
+        'GPS Longitude': '1', 'GPS Latitude': '2',
+        'Audited Outcome 2018/19': 2100.0,
+        'Full Year Forecast 2019/20': 3100.0,
+        'Budget year 2020/21': 4000.0,
+        'Budget year 2021/22': 5000.0,
+        'Budget year 2022/23': 6100.0
+    }
+    yield mock_data
+
+def mock_new_project_row():
+    """Composite key fields should not match mock_project_row() so that this creates a new project"""
+    mock_data = { 'Function': 'Administrative and Corporate Support',
+        'Project Description': 'P-CNIN FURN & OFF EQUIP - NEW DESCRIPTION',
+        'Project Number': 'PC002003005_00002',
+        'Type': 'Renewal',
+        'MTSF Service Outcome': 'A project update',
+        'IUDF': 'Governance',
+        'Own Strategic Objectives': 'TO BE CORRECTED',
+        'Asset Class': 'Furniture and Office',
+        'Asset Sub-Class': 'Equipment',
+        'Ward Location': 'Head Office',
+        'GPS Longitude': '1', 'GPS Latitude': '2',
+        'Audited Outcome 2018/19': 2100.0,
+        'Full Year Forecast 2019/20': 3100.0,
+        'Budget year 2020/21': 4000.0,
+        'Budget year 2021/22': 5000.0,
+        'Budget year 2022/23': 6100.0
+    }
+    yield mock_data
+
+
+def verify_expenditure(this, amount, budget_phase, year):
+    expenditure = Expenditure.objects.get(amount=amount)
+    this.assertEquals(expenditure.budget_phase.name, budget_phase)
+    this.assertEquals(expenditure.financial_year.budget_year, year)
```

Comment:
I'd call this `verify_expenditure_exists` otherwise I have to read the method definition to know what it does.

I'd also include the other fields as get() parameters, otherwise this will fail when more than one expenditure item have the same amount. That's future-proofing and not super urgent now, but an easy improvement.